tests/series/test_utf8_ops.py

```diff
@@ -434,6 +434,57 @@ def test_series_utf8_left_bad_dtype() -> None:
         s.str.left(nchars)
 
 
+@pytest.mark.parametrize(
+    ["data", "nchars", "expected"],
+    [
+        (["foo", "barbaz", "quux"], [0, 1, 2], ["", "z", "ux"]),
+        (["foo", "barbaz", "quux"], [1], ["o", "z", "x"]),
+        (["foo"], [0, 1, 2], ["", "o", "oo"]),
+        ([None], [0, 1, 2], [None, None, None]),
+        (["foo", "barbaz", "quux"], [None], [None, None, None]),
+        ([[], [0, 1], []]),
+        ([["foo"] * 4, [], []]),
+        (["foo", None, "barbaz", "quux"], [0, 1, 1, None], ["", None, "z", None]),
+        ([None] * 4, [1] * 4, [None] * 4),
+        (["foo"] * 4, [None] * 4, [None] * 4),
+        ([None], [1] * 4, [None] * 4),
+        (["foo"] * 4, [None], [None] * 4),
```

Comment:
Can you also add tests with emojis here? Since they are encoded using 4 bytes in UTF-8, it would be good to test that they are not split in the middle.

Reply:
added